src/ConfigCatClient/Override/LocalDictionaryDataSource.cs

```diff
@@ -8,17 +8,25 @@ namespace ConfigCat.Client.Override
 {
     internal sealed class LocalDictionaryDataSource : IOverrideDataSource
     {
-        private readonly IDictionary<string, Setting> settings;
+        private readonly IDictionary<string, Setting> initialSettings;
+        private readonly IDictionary<string, object> overrideValues;
+        private readonly bool watchChanges;
 
-        public LocalDictionaryDataSource(IDictionary<string, object> overrideValues)
+        public LocalDictionaryDataSource(IDictionary<string, object> overrideValues, bool watchChanges)
         {
-            this.settings = overrideValues.ToDictionary(kv => kv.Key, kv => kv.Value.ToSetting());
+            this.initialSettings = overrideValues.ToDictionary(kv => kv.Key, kv => kv.Value.ToSetting());
+            this.overrideValues = overrideValues;
```

Comment:
nitpick: In case changes are not watched, the source dictionary is stored unnecessarily and the `LocalDictionaryDataSource` instance may prevent it from being garbage collected.

```suggestion
            if (watchChanges)
                this.overrideValues = overrideValues;
```

In this case, we don't really need the `watchChanges` field either because the nullability of the `overrideValues` field tells us whether we have a watched source or not. (We could introduce a `WatchChanges` private property to make this obvious.)

Reply:
Done